### PR TITLE
 [Feat #23131] Display status on user page

### PR DESCRIPTION
**Pull request by perunt**
### Fixed Issues

$ https://github.com/Expensify/App/issues/23131
PROPOSAL: 

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

Test Case 1: Status with Emoji + Text
1. Use Account A to create a status (emoji + text).
2. Switch to Account B and open the user details page of Account A.
3. Verify if Account A has the status displayed.

Test Case 2: Status with Only Emoji
1. Use Account A to create a status (only emoji).
2. Switch to Account B and open the user details page of Account A.
3. Verify if Account A has the status displayed.

Test Case 3: Status with Extended Text
1. Use Account A to create a lengthy status.
2. Switch to Account B and open the user details page of Account A.
3. Verify if Account A has the status displayed and if it fits correctly.
4. Check if the tooltip renders correctly.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
Screenshot 2023-08-09 at 18 53 11

</details>

<details>
<summary>Mobile Web - Chrome</summary>
Screenshot 2023-08-09 at 19 04 14

</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2023-08-09 at 19 04 14

</details>

<details>
<summary>Desktop</summary>
Screenshot 2023-08-09 at 19 13 50

</details>

<details>
<summary>iOS</summary>
Screenshot 2023-08-09 at 19 03 22

</details>

<details>
<summary>Android</summary>

Screenshot 2023-08-09 at 19 15 55

videos here -->

</details>

### CODE REVIEW

File: src/pages/settings/Profile/CustomStatus/StatusPage.js [hunk 1]
```diff
@@ -33,16 +33,16 @@ function StatusPage({draftStatus, currentUserPersonalDetails}) {
     const draftText = lodashGet(draftStatus, 'text');
 
     const defaultEmoji = draftEmojiCode || currentUserEmojiCode;
-    const defaultText = draftText || currentUserStatusText;
-    const customStatus = defaultEmoji ? `${defaultEmoji} ${defaultText}` : '';
+    const defaultText = draftEmojiCode ? draftText : currentUserStatusText;
+    const customStatus = draftEmojiCode ? `${draftEmojiCode} ${draftText}` : `${currentUserEmojiCode || ''} ${currentUserStatusText || ''}` ;
```
Comment (perunt) [inline src/pages/settings/Profile/CustomStatus/StatusPage.js:37]: It's a fix for saving an empty status

**Comment by rushatgabhane** [comment]
@perunt could you please add testing steps so that QA can test

**Review by rushatgabhane — APPROVED** [review]

File: src/pages/ProfilePage.js [hunk 2]
```diff
@@ -178,6 +184,20 @@ function ProfilePage(props) {
                                     {displayName}
                                 </Text>
                             )}
+                            {hasStatus ? (
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/ProfilePage.js:187]: Just used a pattern from that screen 😄 
Updated, thanks!

**Review by rushatgabhane — APPROVED** [review]
LGTM

**Comment by perunt** [comment]
hey, @shawnborton need your help 😃 
We use a 'one line' style for items. The status is capped at 100 words, so I added a tooltip for the full message. Should I format it over multiple lines?
Screenshot 2023-08-09 at 18 35 32

**Comment by shawnborton** [comment]
What does this look like without the tooltip? I am thinking we can just use the same pattern as a push-to-page row that can't be edited. That being the case, maybe we don't add the one-line cap but rather enforce a character limit when creating the status.

File: src/pages/ProfilePage.js [hunk 3]
```diff
@@ -178,6 +184,20 @@ function ProfilePage(props) {
                                     {displayName}
                                 </Text>
                             )}
+                            {hasStatus && (
```
Comment (stitesExpensify) [inline src/pages/ProfilePage.js:187]: Let's hide this behind the beta as well (and all other PRs). We don't want internal people setting a status, then having external people see it and be confused what it is.

**Review by stitesExpensify — CHANGES_REQUESTED** [review]
Looking good! One comment and waiting for Shawn's opinion on no tooltip (when those images are posted) before I merge

**Comment by perunt** [comment]
Screenshot 2023-08-09 at 21 41 37
Screenshot 2023-08-09 at 21 41 31

Screenshot 2023-08-09 at 21 42 30
Screenshot 2023-08-09 at 21 42 18

I've noticed some people leave really long 'statuses'. It might make sense to limit the characters to the length of one sentence.

**Comment by shawnborton** [comment]
Yeah, I don't think we should limit the display of the status. So I would remove the one-line truncation you have. And yeah, rather, let's add a character limit to the status itself.

**Comment by stitesExpensify** [comment]
What do you think we should set the limit to? Looks like slack does 100

**Comment by stitesExpensify** [comment]
Not technically related to this PR but can you update this line to be translated please? https://github.com/Expensify/App/pull/24330/files#diff-06be2d0d91e1fd806520ae7745785251dbd0c77d86ee71b828b540cfe42596ccR81

**Comment by perunt** [comment]
Nice, then just summarize. I'll keep the limit to 100 characters. This amount of characters fits in 2 lines on the smallest screen (iPhone SE). The tooltip has been removed. I'll drop the screenshot below
Screenshot 2023-08-10 at 10 21 02
Screenshot 2023-08-10 at 10 43 47

File: src/pages/ProfilePage.js (hunk 3 above)
Comment (perunt) [reply src/pages/ProfilePage.js:187]: 👍🏻

**Review by stitesExpensify — COMMENTED** [review]
Code is looking great but I found one bug, when you are on a user's profile and their status changes, it is not automatically reflected. You need to close the profile and then go back into it to see the update. Can we change it so that when the onyx changes the profile also changes?

**Comment by perunt** [comment]
Hmm, that's weird because it works for me
do we speak about the same profile screen?

https://github.com/Expensify/App/assets/48593211/3546dd17-f566-42c3-ba03-818f5b509ca2

**Comment by stitesExpensify** [comment]
Hm maybe my pusher was just being weird or something. Let me try again

**Comment by stitesExpensify** [comment]
Yep my pusher was just not working locally.

**Review by stitesExpensify — APPROVED** [review]

**Comment by JmillsExpensify** [comment]
@stitesExpensify Do you mind confirming that @rushatgabhane that @rushatgabhane is owed the standard $1k for this PR review?

**Comment by stitesExpensify** [comment]
Yep, standard 1k!

**Comment by JmillsExpensify** [comment]
Thanks so much! This $1k payment is approved for payment via NewDot.
